src/systems/userinput/devices/mouse.js

```diff
@@ -62,8 +62,10 @@ export class MouseDevice {
     const left = event.button === 0;
     const middle = event.button === 1;
     const right = event.button === 2;
-    this.coords[0] = (event.clientX / window.innerWidth) * 2 - 1;
-    this.coords[1] = -(event.clientY / window.innerHeight) * 2 + 1;
+    // Note: This assumes the canvas always starts in the top left.
+    // This works with the current sidebar and toolbar layout.
+    this.coords[0] = (event.clientX / this.canvas.clientWidth) * 2 - 1;
+    this.coords[1] = -(event.clientY / this.canvas.clientHeight) * 2 + 1;
```

Comment:
May need to update similar lines for touchscreens:
```
./src/systems/userinput/devices/app-aware-touchscreen.js:46:      x: (touch.clientX / window.innerWidth) * 2 - 1,
./src/systems/userinput/devices/app-aware-touchscreen.js:174:          (touch.clientX / window.innerWidth) * 2 - 1,
./src/systems/userinput/devices/app-aware-touchscreen.js:234:        (touch.clientX / window.innerWidth) * 2 - 1,
./src/systems/userinput/devices/app-aware-touchscreen.js:355:      frame.setVector2(path.touchCameraDelta, delta[0] / window.innerWidth, delta[1] / window.innerHeight);
```
It probably won't affect phones because the panels are full-screen takeovers but for bigger touchscreens like ipads, not updating it may cause issues.

Reply:
Ah, thanks. I don't actually have a way to test on my iPad due to the secure websocket connection and locally signed cert. I'll make the change and I guess we can test on the staging server?

Reply:
I made the change. Idk if it works or not though. I don't have a way to test it.